src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -115,8 +115,8 @@ private bool IsAnalyzerSuppressedForTree(DiagnosticAnalyzer analyzer, SyntaxTree
             Func<DiagnosticAnalyzer, object?> getAnalyzerGate,
             Func<SyntaxTree, SemanticModel> getSemanticModel,
             bool logExecutionTime = false,
-            Action<Diagnostic, DiagnosticAnalyzer, bool>? addCategorizedLocalDiagnostic = null,
-            Action<Diagnostic, DiagnosticAnalyzer>? addCategorizedNonLocalDiagnostic = null,
+            Action<Diagnostic, DiagnosticAnalyzer, bool, CancellationToken>? addCategorizedLocalDiagnostic = null,
+            Action<Diagnostic, DiagnosticAnalyzer, CancellationToken>? addCategorizedNonLocalDiagnostic = null,
             Action<Suppression>? addSuppression = null,
             CancellationToken cancellationToken = default)
```

Comment:
❓ Why are we still keeping this if we are pooling executor instances and know it's prone to incorrect use?

Reply:
Below code ensures that the pooled executor always uses the latest cancellation token: https://github.com/dotnet/roslyn/blob/6c98e5cb10876d820d74c3eca9a43ef5592b0251/src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs#L700-L702

I am planning to clean this up further in a follow-up refactoring that removes this cancellationToken field and instead passes the token into all the methods within the executor. However, that will be a pure refactoring, while this PR is a targeted bug fix, so I wanted to keep them separate.